Android audio input crashes on Xiaomi Pocophone F1 (even with Record Audio permission)

<!-- Please search existing issues for potential duplicates before filing yours:
https://github.com/godotengine/godot/issues?q=is%3Aissue
-->

**Godot version:**
<!-- Specify commit hash if non-official. -->
[e7dd05f](https://github.com/godotengine/godot/commit/e7dd05fed80d80b328977450bc3a48f500d5cd44), testing merged #26407.

**OS/device including version:**
<!-- Specify GPU model and drivers if graphics-related. -->
 Xiaomi Pocophone F1, Android 9 (MIUI 10)


**Issue description:**
<!-- What happened, and what was expected. -->
After installing the Mic Record demo to my phone, with Record Audio permission enabled in the AndroidManifest.xml, and then enabling the permission manually on the phone for the installed app (for some reason it doesn't ask me for it on install), I get a crash.

The application does run, but crashes after one or two seconds, without me doing anything.

```
03-01 10:38:02.932  1681  1703 I ActivityManager: Displayed org.godotengine.micrecord2/org.godotengine.godot.Godot: +617ms
03-01 10:38:02.932  1681  1703 I Timeline: Timeline: Activity_windows_visible id: ActivityRecord{b3e9e4e u0 org.godotengine.micrecord2/org.godotengine.godot.Godot t174} time:48037771
03-01 10:38:02.933  2925  3239 D PowerKeeper.Event: notifyActivityLaunchTime: org.godotengine.micrecord2/org.godotengine.godot.Godot totalTime: 617
03-01 10:38:02.934  1681  1695 D ActivityTrigger: ActivityTrigger activityStopTrigger 
03-01 10:38:02.934  1902  1902 D RecentsActivity: onStop
03-01 10:38:02.935  1902  1902 D EventBus: [1902, u0] send(RecentsVisibilityChangedEvent)
03-01 10:38:02.935  1902  1902 D EventBus: [1902, u0]  -> RecentsViewTouchHandler [0xfbd74f7, P4] onBusEvent(RecentsVisibilityChangedEvent)
03-01 10:38:02.935  1902  1902 D EventBus: [1902, u0] onBusEvent(RecentsVisibilityChangedEvent) duration: 7 microseconds, avg: 279
03-01 10:38:02.935  1902  1902 D EventBus: [1902, u0]  -> TaskStackView [0x4980223, P3] onBusEvent(RecentsVisibilityChangedEvent)
03-01 10:38:02.935  1902  1902 D EventBus: [1902, u0] unregister()
03-01 10:38:02.935  1902  1902 I chatty  : uid=1000(system) com.android.systemui identical 2 lines
03-01 10:38:02.935  1902  1902 D EventBus: [1902, u0] unregister()
03-01 10:38:02.935  1902  1902 D EventBus: [1902, u0] onBusEvent(RecentsVisibilityChangedEvent) duration: 353 microseconds, avg: 279
03-01 10:38:02.935  1902  1902 D EventBus: [1902, u0]  -> RecentMenuView [0x1a36648, P3] onBusEvent(RecentsVisibilityChangedEvent)
03-01 10:38:02.935  1902  1902 D EventBus: [1902, u0] onBusEvent(RecentsVisibilityChangedEvent) duration: 2 microseconds, avg: 279
03-01 10:38:02.935  1902  1902 D EventBus: [1902, u0]  -> RecentsImpl [0xfd07611, P1] onBusEvent(RecentsVisibilityChangedEvent)
03-01 10:38:02.935  1902  1902 D EventBus: [1902, u0] onBusEvent(RecentsVisibilityChangedEvent) duration: 2 microseconds, avg: 279
03-01 10:38:02.935  1902  1902 D EventBus: [1902, u0]  -> Recents [0x530a65a, P1] onBusEvent(RecentsVisibilityChangedEvent)
03-01 10:38:02.935  1902  1902 D EventBus: [1902, u0] onBusEvent(RecentsVisibilityChangedEvent) duration: 53 microseconds, avg: 279
03-01 10:38:02.946  1681  1766 I WindowManager: Screen frozen for +595ms due to Window{c4327bd u0 Splash Screen org.godotengine.micrecord2}
03-01 10:38:03.000 14338 14378 W libOpenSLES: Conversion from OpenSL ES positional channel mask 0x4 to Android mask 0 loses channels
03-01 10:38:03.000 14338 14378 I libOpenSLES: Emulating old channel mask behavior (ignoring positional mask 0x4, using default mask 0x10 based on channel count of 1)
03-01 10:38:03.000 14338 14378 D AudioRecord: set(): inputSource 0, sampleRate 44100, format 0x1, channelMask 0x10, frameCount 0, notificationFrames 0, sessionId 0, transferType 1, flags 0x5, opPackageName  uid -1, pid -1
03-01 10:38:03.002   714   906 D APM_AudioPolicyManager: getInputForAttr() source 0, samplingRate 44100, format 0x1, channelMask 0x10,session 1769, flags 0x5
03-01 10:38:03.003   654  1824 D audio_hw_primary: adev_open_input_stream: enter: sample_rate(44100) channel_mask(0x10) devices(0x80000004)        stream_handle(0xe7cdb300) io_handle(230) source(1) format 1
03-01 10:38:03.003   654  1824 W audio_hw_utils: audio_extn_utils_update_stream_input_app_type_cfg: App type could not be selected. Falling back to default
03-01 10:38:03.004   714 14383 I AudioFlinger: AudioFlinger's thread 0xe40033c0 tid=14383 ready to run
03-01 10:38:03.004   654  8396 D audio_hw_primary: in_standby: enter: stream (0xe7cdb300) usecase(18: audio-record)
03-01 10:38:03.005   714   906 D APM_AudioPolicyManager: getInputForAttr() returns input 230 type 0 selectedDeviceId 14
03-01 10:38:03.005   714   906 W AudioFlinger: createRecordTrack_l(): mismatch between requested flags (00000005) and input flags (00000001)
03-01 10:38:03.005   714   906 D APM_AudioPolicyManager: releaseInput() 230
03-01 10:38:03.005   714   906 D APM_AudioPolicyManager: closeInput(230)
03-01 10:38:03.006   654  1824 D audio_hw_primary: in_standby: enter: stream (0xe7cdb300) usecase(18: audio-record)
03-01 10:38:03.009   654  1824 D audio_hw_primary: adev_close_input_stream: enter:stream_handle(0xe7cdb300)
03-01 10:38:03.009   654  1824 D msm8974_platform: platform_set_echo_reference: disabling echo-reference voip-headphones
03-01 10:38:03.009   654  1824 D audio_hw_primary: in_standby: enter: stream (0xe7cdb300) usecase(18: audio-record)
03-01 10:38:03.011   714   906 D APM_AudioPolicyManager: getInputForAttr() source 0, samplingRate 44100, format 0x1, channelMask 0x10,session 1769, flags 0
03-01 10:38:03.011   654  1824 D audio_hw_primary: adev_open_input_stream: enter: sample_rate(44100) channel_mask(0x10) devices(0x80000004)        stream_handle(0xe7cdb300) io_handle(238) source(1) format 1
03-01 10:38:03.011   654  1824 W audio_hw_utils: audio_extn_utils_update_stream_input_app_type_cfg: App type could not be selected. Falling back to default
03-01 10:38:03.012   714   906 D APM_AudioPolicyManager: getInputForAttr() returns input 238 type 0 selectedDeviceId 14
03-01 10:38:03.012   714 14384 I AudioFlinger: AudioFlinger's thread 0xe4003340 tid=14384 ready to run
03-01 10:38:03.012   654   897 D audio_hw_primary: in_standby: enter: stream (0xe7cdb300) usecase(18: audio-record)
03-01 10:38:03.014   654  1822 D audio_hw_primary: in_standby: enter: stream (0xe7cdb300) usecase(18: audio-record)
03-01 10:38:03.016   714   907 V AudioPolicyManagerCustom: startInput(input:238, session:1769, silenced:0, concurrency:0)
03-01 10:38:03.017   714   907 V AudioPolicyManagerCustom: startInput recording for org.godotengine.micrecord2  uid 10208
03-01 10:38:03.018   654  1822 D audio_hw_primary: in_set_parameters: enter: kvpairs=bottom=;input_source=1;routing=-2147483644
03-01 10:38:03.019   714   907 V AudioPolicyManagerCustom: AudioPolicyManager::startInput() input source = 1
03-01 10:38:03.020   654 14385 D audio_hw_primary: start_input_stream: enter: stream(0xe7cdb300)usecase(18: audio-record)
03-01 10:38:03.020   654 14385 D audio_hw_primary: select_devices for use case (audio-record)
03-01 10:38:03.020   654 14385 V audio_hw_primary: voice_extn_compress_voip_is_active: COMPRESS_VOIP_ENABLED is not defined
03-01 10:38:03.020   654 14385 D audio_hw_primary: select_devices: out_snd_device(0: ) in_snd_device(74: handset-mic)
03-01 10:38:03.020   654 14385 I msm8974_platform: platform_check_and_set_capture_codec_backend_cfg:txbecf: afe: bitwidth 16, samplerate 44100, channel 1 format 1, backend_idx 7 usecase = 18 device (handset-mic)
03-01 10:38:03.020   654 14385 I msm8974_platform: platform_check_capture_codec_backend_cfg:txbecf: afe: Codec selected backend: 7 current bit width: 16 and sample rate: 44100, channels 1 format 1
03-01 10:38:03.020   654 14385 W msm8974_platform: platform_get_backend_index:napb: Invalid device - 0 
03-01 10:38:03.020   654 14385 I msm8974_platform: platform_check_capture_codec_backend_cfg:txbecf: afe: Codec selected backend: 7 updated bit width: 16 and sample rate: 44100
03-01 10:38:03.020   654 14385 D audio_hw_primary: check_usecases_capture_codec_backend:becf: force routing 0
03-01 10:38:03.020   654 14385 D hardware_info: hw_info_append_hw_type : device_name = handset-mic
03-01 10:38:03.020   654 14385 D msm8974_platform: platform_split_snd_device: snd_device(74) num devices(0) new_snd_devices(0)
03-01 10:38:03.020   654 14385 D audio_hw_primary: enable_snd_device: snd_device(74: handset-mic)
03-01 10:38:03.020   654 14385 I soundtrigger: audio_extn_sound_trigger_update_device_status: device 0x4a of type 1 for Event 1, with Raise=0
03-01 10:38:03.020   654 14385 D audio_route: Apply path: handset-mic
03-01 10:38:03.021  2925  3239 I PowerCheckerService: uid:10208 change to active
03-01 10:38:03.022  2925  3069 I AppActiveChecker: uid:10208 change to active
03-01 10:38:03.024   654 14385 W audio_hw_utils: audio_extn_utils_update_stream_input_app_type_cfg: App type could not be selected. Falling back to default
03-01 10:38:03.024   654 14385 D soundtrigger: audio HAL using same device 74 as ST
03-01 10:38:03.024   654 14385 D soundtrigger: audio_extn_sound_trigger_update_stream_status: uc_info->id 18 of type 1 device 0x00 for Event 3, with Raise=1
03-01 10:38:03.024   654 14385 W soundtrigger: audio_extn_sound_trigger_update_stream_status: device[0x00]: speaker is NOT used, do NOT transit to adsp. 
03-01 10:38:03.024   654 14385 D audio_hw_utils: audio_extn_utils_send_app_type_cfg: usecase->in_snd_device handset-mic
03-01 10:38:03.024   654 14385 D msm8974_platform: platform_split_snd_device: snd_device(74) num devices(0) new_snd_devices(0)
03-01 10:38:03.024   654 14385 D audio_hw_utils: audio_extn_btsco_get_sample_rate:Not a BT SCO device, need not update sampling rate
03-01 10:38:03.024   654 14385 I audio_hw_utils: send_app_type_cfg_for_device CAPTURE app_type 69938, acdb_dev_id 4, sample_rate 48000, snd_device_be_idx 3
03-01 10:38:03.024   654 14385 D msm8974_platform: platform_split_snd_device: snd_device(74) num devices(1) new_snd_devices(0)
03-01 10:38:03.024   654 14385 I ACDB-LOADER: ACDB -> send_audio_cal, acdb_id = 4, path = 1, app id = 0x11132, sample rate = 48000
03-01 10:38:03.024   654 14385 D         : Failed to fetch the lookup information of the device 00000004 
03-01 10:38:03.024   654 14385 E ACDB-LOADER: Error: ACDB AudProc vol returned = -19
03-01 10:38:03.024   654 14385 I ACDB-LOADER: ACDB -> GET_AFE_TOPOLOGY_ID for adcd_id 4, Topology Id 112fb
03-01 10:38:03.024   654 14385 D         : Failed to fetch the lookup information of the device 00000004 
03-01 10:38:03.024   654 14385 E ACDB-LOADER: Error: ACDB_CMD_GET_AFE_COMMON_TABLE_SIZE Returned = -19
03-01 10:38:03.024   654 14385 D         : Failed to fetch the lookup information of the device 00000004 
03-01 10:38:03.024   654 14385 E ACDB-LOADER: Error: ACDB AFE returned = -19
03-01 10:38:03.024   654 14385 D audio_hw_primary: enable_audio_route: apply mixer and update path: audio-record
03-01 10:38:03.024   654 14385 D audio_route: Apply path: audio-record
03-01 10:38:03.024   654 14385 D audio_hw_primary: select_devices: done
03-01 10:38:03.066   654 14385 D audio_hw_primary: start_input_stream: exit
03-01 10:38:04.556  1681 11035 D PerfShielderService: com.android.systemui|StatusBar|262|48037494172299|104|0|5
03-01 10:38:04.557  2704  2724 W MiuiPerfServiceClient: interceptAndQueuing:1902|com.android.systemui|262|104|unknown|null|StatusBar|48037494172299|Slow main thread|5
03-01 10:38:05.120  1681  1703 I system_server: Explicit concurrent copying GC freed 31608(1889KB) AllocSpace objects, 1(20KB) LOS objects, 42% free, 24MB/42MB, paused 296us total 164.700ms
03-01 10:38:08.380 14338 14378 E godot   : **ERROR**: FATAL: Index p_index=512 out of size (size()=0)
03-01 10:38:08.380 14338 14378 E godot   :    At: ./core/cowdata.h:151:get() - FATAL: Index p_index=512 out of size (size()=0)
03-01 10:38:08.381 14338 14378 F libc    : Fatal signal 4 (SIGILL), code 1 (ILL_ILLOPC), fault addr 0xd1194a74 in tid 14378 (AudioTrack), pid 14338 (gine.micrecord2)
03-01 10:38:08.436 14392 14392 I crash_dump32: obtaining output fd from tombstoned, type: kDebuggerdTombstone
03-01 10:38:08.436  1023  1023 I /system/bin/tombstoned: received crash request for pid 14378
03-01 10:38:08.438 14392 14392 I crash_dump32: performing dump of process 14338 (target tid = 14378)
03-01 10:38:08.445 14392 14392 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
03-01 10:38:08.445 14392 14392 F DEBUG   : Build fingerprint: 'Xiaomi/beryllium/beryllium:9/PKQ1.180729.001/V10.2.2.0.PEJMIXM:user/release-keys'
03-01 10:38:08.445 14392 14392 F DEBUG   : Revision: '0'
03-01 10:38:08.445 14392 14392 F DEBUG   : ABI: 'arm'
03-01 10:38:08.445 14392 14392 F DEBUG   : pid: 14338, tid: 14378, name: AudioTrack  >>> org.godotengine.micrecord2 <<<
03-01 10:38:08.445 14392 14392 F DEBUG   : signal 4 (SIGILL), code 1 (ILL_ILLOPC), fault addr 0xd1194a74
03-01 10:38:08.446 14392 14392 F DEBUG   :     r0  cf7fefe8  r1  98419eb6  r2  00000008  r3  40000000
03-01 10:38:08.446 14392 14392 F DEBUG   :     r4  80000000  r5  b8000000  r6  cf7ff8a0  r7  d112f1fc
03-01 10:38:08.446 14392 14392 F DEBUG   :     r8  cfec1960  r9  f07ae251  r10 ef05ab78  r11 cf7ff060
03-01 10:38:08.446 14392 14392 F DEBUG   :     ip  ef5e7d00  sp  cf7ff010  lr  98419eb6  pc  d1194a74
03-01 10:38:08.452 14338 14378 F libc    : failed to wait for crash_dump helper: No child processes
03-01 10:38:08.508 14392 14392 F DEBUG   : 
03-01 10:38:08.508 14392 14392 F DEBUG   : backtrace:
03-01 10:38:08.508 14392 14392 F DEBUG   :     #00 pc 00105a74  /data/app/org.godotengine.micrecord2-mnpqpDi9rTpufKTsQaxPNA==/lib/arm/libgodot_android.so (CowData<int>::get(int) const+272)
03-01 10:38:08.508 14392 14392 F DEBUG   :     #01 pc 00100dd0  /data/app/org.godotengine.micrecord2-mnpqpDi9rTpufKTsQaxPNA==/lib/arm/libgodot_android.so (Vector<int>::operator[](int) const+32)
03-01 10:38:08.508 14392 14392 F DEBUG   :     #02 pc 017bf248  /data/app/org.godotengine.micrecord2-mnpqpDi9rTpufKTsQaxPNA==/lib/arm/libgodot_android.so (AudioStreamPlaybackMicrophone::_mix_internal(AudioFrame*, int)+608)
03-01 10:38:08.508 14392 14392 F DEBUG   :     #03 pc 017be810  /data/app/org.godotengine.micrecord2-mnpqpDi9rTpufKTsQaxPNA==/lib/arm/libgodot_android.so (AudioStreamPlaybackResampled::mix(AudioFrame*, float, int)+1696)
03-01 10:38:08.508 14392 14392 F DEBUG   :     #04 pc 017bf850  /data/app/org.godotengine.micrecord2-mnpqpDi9rTpufKTsQaxPNA==/lib/arm/libgodot_android.so (AudioStreamPlaybackMicrophone::mix(AudioFrame*, float, int)+52)
03-01 10:38:08.508 14392 14392 F DEBUG   :     #05 pc 011ab2e4  /data/app/org.godotengine.micrecord2-mnpqpDi9rTpufKTsQaxPNA==/lib/arm/libgodot_android.so (AudioStreamPlayer::_mix_internal(bool)+212)
03-01 10:38:08.508 14392 14392 F DEBUG   :     #06 pc 011ab7f0  /data/app/org.godotengine.micrecord2-mnpqpDi9rTpufKTsQaxPNA==/lib/arm/libgodot_android.so (AudioStreamPlayer::_mix_audio()+392)
03-01 10:38:08.508 14392 14392 F DEBUG   :     #07 pc 011abb58  /data/app/org.godotengine.micrecord2-mnpqpDi9rTpufKTsQaxPNA==/lib/arm/libgodot_android.so (AudioStreamPlayer::_mix_audios(void*)+20)
03-01 10:38:08.508 14392 14392 F DEBUG   :     #08 pc 01521e7c  /data/app/org.godotengine.micrecord2-mnpqpDi9rTpufKTsQaxPNA==/lib/arm/libgodot_android.so (AudioServer::_mix_step()+580)
03-01 10:38:08.508 14392 14392 F DEBUG   :     #09 pc 01520c88  /data/app/org.godotengine.micrecord2-mnpqpDi9rTpufKTsQaxPNA==/lib/arm/libgodot_android.so (AudioServer::_driver_process(int, int*)+136)
03-01 10:38:08.508 14392 14392 F DEBUG   :     #10 pc 01520b74  /data/app/org.godotengine.micrecord2-mnpqpDi9rTpufKTsQaxPNA==/lib/arm/libgodot_android.so (AudioDriver::audio_server_process(int, int*, bool)+92)
03-01 10:38:08.508 14392 14392 F DEBUG   :     #11 pc 000a0078  /data/app/org.godotengine.micrecord2-mnpqpDi9rTpufKTsQaxPNA==/lib/arm/libgodot_android.so (AudioDriverOpenSL::_buffer_callback(SLAndroidSimpleBufferQueueItf_ const* const*)+180)
03-01 10:38:08.508 14392 14392 F DEBUG   :     #12 pc 000a0220  /data/app/org.godotengine.micrecord2-mnpqpDi9rTpufKTsQaxPNA==/lib/arm/libgodot_android.so (AudioDriverOpenSL::_buffer_callbacks(SLAndroidSimpleBufferQueueItf_ const* const*, void*)+36)
03-01 10:38:08.508 14392 14392 F DEBUG   :     #13 pc 0000bdf7  /system/lib/libwilhelm.so (audioTrack_callBack_pullFromBuffQueue(int, void*, void*)+114)
03-01 10:38:08.508 14392 14392 F DEBUG   :     #14 pc 00035d1d  /system/lib/libaudioclient.so (android::AudioTrack::processAudioBuffer()+1716)
03-01 10:38:08.508 14392 14392 F DEBUG   :     #15 pc 00037739  /system/lib/libaudioclient.so (android::AudioTrack::AudioTrackThread::threadLoop()+128)
03-01 10:38:08.508 14392 14392 F DEBUG   :     #16 pc 0000c107  /system/lib/libutils.so (android::Thread::_threadLoop(void*)+290)
03-01 10:38:08.508 14392 14392 F DEBUG   :     #17 pc 000637f5  /system/lib/libc.so (__pthread_start(void*)+22)
03-01 10:38:08.508 14392 14392 F DEBUG   :     #18 pc 0001e019  /system/lib/libc.so (__start_thread+24)
03-01 10:38:08.921 14392 14392 E crash_dump32: cannot open libmiuindbg.so: No such file or directory
03-01 10:38:08.922  1023  1023 E /system/bin/tombstoned: Tombstone written to: /data/tombstones/tombstone_09
03-01 10:38:08.924  1681 14395 W ActivityManager:   Force finishing activity org.godotengine.micrecord2/org.godotengine.godot.Godot
```


**Steps to reproduce:**
****
- Clone `mic_record` official demo: https://github.com/godotengine/godot-demo-projects/tree/master/audio/mic_record
- Add Android preset, specify identifier and name, and Record Audio permission
- Deploy to phone with one-click deploy
- Go to App Info for the app, and enable Microphone permission
- Run the game
- See crash and check `adb logcat`

**Minimal reproduction project:**
<!-- Recommended as it greatly speeds up debugging. Drag and drop a zip archive to upload it. -->
 https://github.com/godotengine/godot-demo-projects/tree/master/audio/mic_record
[mic_record.zip](https://github.com/godotengine/godot/files/2919284/mic_record.zip) (demo with export preset pre-configured)

Testing. FYI, it raises a `-Wsign-compare` warning:
```
servers/audio/audio_stream.cpp: In member function 'virtual void AudioStreamPlaybackMicrophone::_mix_internal(AudioFrame*, int)':
servers/audio/audio_stream.cpp:153:44: warning: comparison of integer expressions of different signedness: 'unsigned int' and 'int' [-Wsign-compare]
    if (input_size > input_ofs && input_ofs < buf.size()) {
                                  ~~~~~~~~~~^~~~~~~~~~~~
```

--- a/servers/audio/audio_stream.cpp
+++ b/servers/audio/audio_stream.cpp
@@ -150,7 +150,7 @@ void AudioStreamPlaybackMicrophone::_mix_internal(AudioFrame *p_buffer, int p_fr
 		input_ofs = 0;
 	} else {
 		for (int i = 0; i < p_frames; i++) {
-			if (input_size > input_ofs) {
+			if (input_size > input_ofs && (int)input_ofs < buf.size()) {
 				float l = (buf[input_ofs++] >> 16) / 32768.f;
 				if ((int)input_ofs >= buf.size()) {
 					input_ofs = 0;

--- a/servers/audio_server.cpp
+++ b/servers/audio_server.cpp
@@ -90,12 +90,16 @@ void AudioDriver::input_buffer_init(int driver_buffer_frames) {
 
 void AudioDriver::input_buffer_write(int32_t sample) {
 
-	input_buffer.write[input_position++] = sample;
-	if ((int)input_position >= input_buffer.size()) {
-		input_position = 0;
-	}
-	if ((int)input_size < input_buffer.size()) {
-		input_size++;
+	if ((int)input_position < input_buffer.size()) {
+		input_buffer.write[input_position++] = sample;
+		if ((int)input_position >= input_buffer.size()) {
+			input_position = 0;
+		}
+		if ((int)input_size < input_buffer.size()) {
+			input_size++;
+		}
+	} else {
+		WARN_PRINTS("input_buffer_write: Invalid input_position=" + itos(input_position) + " input_buffer.size()=" + itos(input_buffer.size()));
 	}
 }
 
@@ -145,6 +149,8 @@ AudioDriver::AudioDriver() {
 
 	_last_mix_time = 0;
 	_mix_amount = 0;
+	input_position = 0;
+	input_size = 0;
 
 #ifdef DEBUG_ENABLED
 	prof_time = 0;

